src/main/java/com/aws/greengrass/clientdevices/auth/metrics/handlers/ServiceErrorEventHandler.java

```diff
@@ -0,0 +1,46 @@
+/*
+ * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package com.aws.greengrass.clientdevices.auth.metrics.handlers;
+
+import com.aws.greengrass.clientdevices.auth.api.DomainEvents;
+import com.aws.greengrass.clientdevices.auth.api.ServiceErrorEvent;
+import com.aws.greengrass.clientdevices.auth.metrics.ClientDeviceAuthMetrics;
+import com.aws.greengrass.logging.api.Logger;
+import com.aws.greengrass.logging.impl.LogManager;
+
+import java.util.function.Consumer;
+import javax.inject.Inject;
+
+public class ServiceErrorEventHandler implements Consumer<ServiceErrorEvent> {
+    private final DomainEvents domainEvents;
+    private final ClientDeviceAuthMetrics metrics;
+    private static final Logger logger = LogManager.getLogger(ServiceErrorEventHandler.class);
+
+    /**
+     * Create handler for service error events.
+     *
+     * @param domainEvents Domain event router
+     * @param metrics      Client Device Auth metrics
+     */
+    @Inject
+    public ServiceErrorEventHandler(DomainEvents domainEvents, ClientDeviceAuthMetrics metrics) {
+        this.domainEvents = domainEvents;
+        this.metrics = metrics;
+    }
+
+    /**
+     * Listen for service error events.
+     */
+    public void listen() {
+        domainEvents.registerListener(this, ServiceErrorEvent.class);
+    }
+
+    @Override
+    public void accept(ServiceErrorEvent event) {
+        metrics.incrementServiceError();
+        logger.atError().log(event.getErrorMessage());
```

Comment:
erroring code already throws exceptions or (should) log, so this wouldn't be providing new information, i think we can skip this log

Reply:
An issue I noticed is that if I don't want to log anything in the handler, then the `ServiceErrorEvent` class wouldn't need any fields since it's only used to call `metrics.incrementServiceError()`. And at that point I don't think we need an event or handler since it would make more sense to just call the increment function directly, so would it be better to use that approach or stay consistent with the other metrics?